### PR TITLE
Harden RequestRateLimiter Tests

### PR DESCRIPTION
**Pull request by atris**
This commit adds higher data size and load in the test path. Also improves
the asserts that are performed.

### CODE REVIEW

File: solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java [hunk 1]
```diff
@@ -66,15 +66,11 @@ public void testConcurrentQueries() throws Exception {
 
     solrDispatchFilter.replaceRateLimitManager(rateLimitManager);
 
-    processTest(client);
+    processTest(client, 10000 /* number of documents */, 350 /* number of queries */);
```
Comment (madrob) [inline solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java:69]: Can we limit the higher footprint to Nightly?

File: solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java [hunk 2]
```diff
@@ -66,15 +66,11 @@ public void testConcurrentQueries() throws Exception {
...
 
     MockRequestRateLimiter mockQueryRateLimiter = (MockRequestRateLimiter) rateLimitManager.getRequestRateLimiter(SolrRequest.SolrRequestType.QUERY);
 
-    assertEquals(25, mockQueryRateLimiter.incomingRequestCount.get());
-    assertTrue("Incoming accepted new request count did not match. Expected 5 incoming " + mockQueryRateLimiter.acceptedNewRequestCount.get(),
```
Comment (madrob) [inline solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java:74]: Somewhat concerning that the fix to the test is to relax the assertion conditions

File: solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java [hunk 3]
```diff
@@ -124,12 +120,12 @@ private void processTest(CloudSolrClient client) throws Exception {
...
 
     try {
-      for (int i = 0; i < 25; i++) {
+      for (int i = 0; i < numQueries; i++) {
         callableList.add(() -> {
           try {
             QueryResponse response = client.query(new SolrQuery("*:*"));
 
-            assertEquals(100, response.getResults().getNumFound());
+            assertEquals(10000, response.getResults().getNumFound());
```
Comment (madrob) [inline solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java:128]: should be numDocuments

File: solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java (hunk 2 above)
Comment (atris) [reply solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java:74]: It isnt really a relaxation -- the remaining assert should cover for all cases that can happen for rate limiting. The catch is that rate limiting is not a guaranteed phenomenon -- we create a high load and it should happen. I have added an additional assert -- let me know if it looks fine.

File: solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java [hunk 4]
```diff
@@ -48,7 +47,7 @@ public static void setupCluster() throws Exception {
     configureCluster(1).addConfig(FIRST_COLLECTION, configset("cloud-minimal")).configure();
   }
 
-  @Test
+  @Nightly
```
Comment (madrob) [inline solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java:50]: This isn't what I meant, sorry for being unclear. Keep this as `@Test` but when selecting the number of documents and queries do something like https://github.com/apache/lucene-solr/blob/master/lucene/core/src/test/org/apache/lucene/index/TestMultiDocValues.java#L52

File: solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java [hunk 5]
```diff
@@ -66,20 +65,19 @@ public void testConcurrentQueries() throws Exception {
...
     MockRequestRateLimiter mockQueryRateLimiter = (MockRequestRateLimiter) rateLimitManager.getRequestRateLimiter(SolrRequest.SolrRequestType.QUERY);
 
-    assertEquals(25, mockQueryRateLimiter.incomingRequestCount.get());
-    assertTrue("Incoming accepted new request count did not match. Expected 5 incoming " + mockQueryRateLimiter.acceptedNewRequestCount.get(),
-        mockQueryRateLimiter.acceptedNewRequestCount.get() < 25);
-    assertTrue("Incoming rejected new request count did not match. Expected 20 incoming " + mockQueryRateLimiter.rejectedRequestCount.get(),
-        mockQueryRateLimiter.rejectedRequestCount.get() > 0);
+    assertEquals(350, mockQueryRateLimiter.incomingRequestCount.get());
+
+    assertTrue((mockQueryRateLimiter.acceptedNewRequestCount.get() == mockQueryRateLimiter.incomingRequestCount.get()
```
Comment (madrob) [inline solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java:74]: Should we assert that accepted + rejected = total?
And that accepted > 0.

Comment (atris) [reply solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java:74]: That is what we do in this assert?
    assertEquals(mockQueryRateLimiter.incomingRequestCount.get(),
        mockQueryRateLimiter.acceptedNewRequestCount.get() + mockQueryRateLimiter.rejectedRequestCount.get());

File: solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java (hunk 4 above)
Comment (madrob) [reply solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java:50]: s/Nightly/Test

File: solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java (hunk 5 above)
Comment (madrob) [reply solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java:77]: yea I was looking at the wrong side of the diff

File: solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java (hunk 4 above)
Comment (atris) [reply solr/core/src/test/org/apache/solr/servlet/TestRequestRateLimiter.java]: Updated

**Pull request merged**